src/control/server/mgmt_system_test.go

```diff
@@ -1001,3 +1001,55 @@ func TestServer_MgmtSvc_getPeerListenAddr(t *testing.T) {
 		})
 	}
 }
+
+func TestServer_MgmtSvc_GetAttachInfo(t *testing.T) {
+	const (
+		ETHER      = 1
+		INFINIBAND = 32
+	)
+
+	for name, tc := range map[string]struct {
+		mgmtSvc          *mgmtSvc
+		clientNetworkCfg *ClientNetworkCfg
+		req              *mgmtpb.GetAttachInfoReq
+		expResp          *mgmtpb.GetAttachInfoResp
+	}{
+		"Server uses verbs + Infiniband": {
+			clientNetworkCfg: &ClientNetworkCfg{Provider: "ofi+verbs", CrtCtxShareAddr: 1, CrtTimeout: 10, NetDevClass: INFINIBAND},
+			req:              &mgmtpb.GetAttachInfoReq{},
+			expResp:          &mgmtpb.GetAttachInfoResp{Provider: "ofi+verbs", CrtCtxShareAddr: 1, CrtTimeout: 10, NetDevClass: INFINIBAND},
+		},
+		"Server uses sockets + Ethernet": {
+			clientNetworkCfg: &ClientNetworkCfg{Provider: "ofi+sockets", CrtCtxShareAddr: 0, CrtTimeout: 5, NetDevClass: ETHER},
+			req:              &mgmtpb.GetAttachInfoReq{},
+			expResp:          &mgmtpb.GetAttachInfoResp{Provider: "ofi+sockets", CrtCtxShareAddr: 0, CrtTimeout: 5, NetDevClass: ETHER},
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			log, buf := logging.NewTestLogger(t.Name())
+			defer common.ShowBufferOnFailure(t, buf)
+			harness := NewIOServerHarness(log)
+			srv := newTestIOServer(log, true)
+
+			if err := harness.AddInstance(srv); err != nil {
+				panic(err)
```

Comment:
panic seems a little heavy-handed here even though it's just a test. Why not `t.Fatal(err.Error())`? That would let other tests run.

Reply:
I see.  I looked at prior art in TestServer_MgmtSvc_PoolCreate which used panic(err), but maybe that wasn't the best representative sample of ways to do this.  I've changed it.